Ref #18: Fixes quality selector options autohide inconsistency on non…

hack4mer opened the pull request:
…-ios mobiles

jthomerson commented:
@hack4mer please:

 1) squash both your commits into one (see https://github.com/silvermine/silvermine-info/blob/master/commit-history.md)
 2) do not include the `package-lock.json` change into that commit - it's not relevant to this work (@onebytegone we should add that one as a separate commit - not sure how we missed doing that before)

Thanks!

@onebytegone run with it after that.

hack4mer commented:
Hi, sorry about that. I have made the requested changes
- Squashed both the commits into one
- Removed `package-lock.json` from the commit

I have also created another commit in a new branch that updates the  `package-lock.json` file. I will open a new PR if you say after this one has been reviewed and merged

onebytegone commented:
Thanks for updating the PR @hack4mer! Unless @jthomerson says otherwise, don't worry about a PR for `package-lock.json`. There is some NPM "housekeeping" that we need to do, update dependencies and such. It'll be easiest to add the lock file in that update.

hack4mer commented:
Sure, thank you @onebytegone. :)

onebytegone commented on src/js/components/QualitySelector.js:
```diff
@@ -63,6 +65,15 @@ module.exports = function(videojs) {
+       */
+      hideQualitySelectorOptions: function() {
+         this.unpressButton();
+         this.player().removeClass('vjs-workinghover');
```
Hmmmm, it is a little concerning that this is removing a class from the player without checking if the class should actually be there. The player only add [this class](https://github.com/videojs/video.js/blob/b25f024a04aacd4151ea49650b32cbe0b722d542/src/js/player.js#L465-L468), on initialization. This line will thereby change the _player's_ styling once the user changes quality. Before the quality is changed [menu buttons will respond to hover](https://github.com/OwenEdwards/video.js/blob/10ef77107b4ddc51a9cfacc88cd5691bf8cec214/src/css/components/menu/_menu-popup.scss#L22-L25). After quality change, this hover behavior will no longer work.

hack4mer replied:
I have added the steps to reproduce the issue on Android phones.

And I get your concern, removing the class works without any issues (for me) but does not seem like a good idea.

Instead of removing this class, should we add another class just to override that one CSS property?

RyanBertrand replied:
The side effect with removing the vjs-workinghover class is it prevents the menus from showing on hover (desktop browsers).

jthomerson commented:
For the reason @onebytegone and @RyanBertrand mention, I'm closing this.
